src/main/java/buildcraftAdditions/api/networking/ISyncObject.java

```diff
@@ -11,7 +11,7 @@
  */
 public interface ISyncObject {
 
-	public ByteBuf writeToByteBuff(ByteBuf buf);
+	void writeToByteBuff(ByteBuf buf);
 
-	public ByteBuf readFromByteBuff(ByteBuf buf);
+	void readFromByteBuff(ByteBuf buf);
```

Comment:
it's clear you didn't test this, this won't work at all, if you don't return the bytebuff it will be reading wrong values